utils/soft_get.py

```diff
@@ -0,0 +1,58 @@
+from difflib import SequenceMatcher
+
+
+def soft_get(obj, field_base_name, dict_=False, case_sensetive=False, best_match=True):
+    """
+    This function used for cases that we want to get some attribute that we
+    either know only few parts of its name or want to prevent from case issues.
+
+    Example:
+        Imagine you have a relationships table and you want to get 'image' field.
+        Since sometimes the exact name of the field is changing among versions, pages, etc.
+        it could be appear as 'Images', 'Image', 'Container Images', Containers Images', etc.
+        Since we don't care for the exact name and know that 'image' is a unique in the table,
+        we can use this function to prevent from this complexity.
+
+    Args:
+        * obj: The object which we want to get the attribute
+        * field_base_name: The base name, a string that we know
+                           for sure that is a sub-string of the target field
+        * dict_: Whether this is a dict and we want to perform the same functionality on its keys
```

Comment:
I think we could do that detection ourselves, by checking isinstance(collections.Mapping) ...